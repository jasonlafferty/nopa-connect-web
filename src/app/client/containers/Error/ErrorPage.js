import React from 'react';
import { Layout } from '../../components';

const errors = {
  '500': {
    title: 'Server error!',
    desc: 'We have been notified about this error'
  },
  '404': {
    title: 'Page not found!',
    desc: 'This page is either no longer available or you have entered a wrong path.'
  }
};

class ErrorPage extends React.Component {

  constructor(props) {
    super(props);
    this.printError = this.printError.bind(this);
  }

  printError (errCode) {
    const error = Object.keys(errors).indexOf(errCode) > -1 ? errors[errCode] : errors['500'];
    return (
      <div>
        <h1>{error.title}</h1>
        <div>{error.desc}</div>
      </div>
    );
  }

  render () {
    return (
      <Layout>
        {this.printError(this.props.route.error)}
      </Layout>
    );
  }
}

ErrorPage.propTypes = {
  route: React.PropTypes.object.isRequired
};

export default ErrorPage;

