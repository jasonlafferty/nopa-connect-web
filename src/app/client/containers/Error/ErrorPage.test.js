import React from 'react';
import ErrorPage from './ErrorPage';
import renderer from 'react-test-renderer';

test('Renders correctly', () => {
  const tree = renderer.create(
    <ErrorPage route={{ error: '404' }} />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
