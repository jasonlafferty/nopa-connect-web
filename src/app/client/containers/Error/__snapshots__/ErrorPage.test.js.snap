// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div>
  <div />
  <div
    className="main-content"
  >
    <div>
      <h1>
        Page not found!
      </h1>
      <div>
        This page is either no longer available or you have entered a wrong path.
      </div>
    </div>
  </div>
  <div />
</div>
`;
